### PR TITLE
[Author] [New] Add js-popup

### PR DESCRIPTION
**Pull request by exelban**
Pull request for issue: #10972 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/exelban/js-popup
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/js-popup
 * License and its reference: Apache License 2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 0
   - Count of stars: 0
   - Count of forks: 0
 * NPM download stats (optional):
   - Downloads in the last day: 20
   - Downloads in the last week: 20
   - Downloads in the last month: 20

# Essential checklist
 * [x] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
I'll squash these all commits now.

**Comment by exelban** [comment]
@PeterDaveHello thanks, but build failed(

**Comment by PeterDaveHello** [comment]
@exelban no worries, I cancelled it, will triger it again when commits squashed, then you can see if it failed and how can you fix it, thanks!

**Comment by exelban** [comment]
@PeterDaveHello ok. Thank you for your help.

**Comment by PeterDaveHello** [comment]
@exelban you're welcome.

**Comment by PeterDaveHello** [comment]
Commit squashed, let's see the test result.

**Comment by exelban** [comment]
@PeterDaveHello I must to add test to package.json?

**Comment by PeterDaveHello** [comment]
@exelban nope, the CI is now running, just give it 2 mins.

**Comment by exelban** [comment]
@PeterDaveHello npm ERR! Test failed.  See above for more details.

File: ajax/libs/js-popup/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "email": "mitrovtsiy@ukr.net"
+  },
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/exelban/js-popup.git"
+  },
+  "bugs": {
+    "url": "https://github.com/exelban/js-popup/issues"
+  },
```
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: L37~L39 should remove.

File: ajax/libs/js-popup/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  "homepage": "https://github.com/exelban/js-popup"
```
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: Homepage doesn't need to be set if it's the same as repository

File: ajax/libs/js-popup/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "js-popup",
+  "npmName": "js-popup",
```
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: Need add `npmFileMap` under npmName
ref: https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

File: ajax/libs/js-popup/package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  "filename": "js-popup.min.js",
+  "version": "1.0.0",
+  "description": "Simple popup messages for JavaScript/ReactJS.",
+  "homepage": "https://github.com/exelban/js-popup/",
```
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: Homepage doesn't need to be set if it's the same as repository

**Comment by PeterDaveHello** [comment]
@sufuf3 the PR already been updated and most of your reviews are for the old ... please check it again.

File: ajax/libs/js-popup/package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "js-popup",
+  "filename": "js-popup.min.js",
+  "version": "1.0.0",
+  "description": "Simple popup messages for JavaScript/ReactJS.",
+  "homepage": "https://github.com/exelban/js-popup/",
```
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: Homepage doesn't need to be set if it's the same as repository

**Comment by exelban** [comment]
@PeterDaveHello passed all CI tests. What now?

**Comment by PeterDaveHello** [comment]
@exelban we'll review it.

File: ajax/libs/js-popup/package.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/exelban/js-popup.git"
+  },
+  "npmName": "js-popup",
+  "npmFileMap": [
+    {
+      "basePath": "release",
```
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: `"basePath": "release",` --> `"basePath": "lib",`

**Comment by exelban** [comment]
@PeterDaveHello ok. Thanks.

File: ajax/libs/js-popup/package.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      "files": [
+        "*"
```
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: `"*"`  --> `"**/*"`

File: ajax/libs/js-popup/package.json (hunk 6 above)
Comment (PeterDaveHello) [inline ajax/libs/js-popup/package.json]: Reason? @sufuf3 are your trained to do things without reason?

**Comment by sufuf3** [comment]
@exelban please fix as [ajax/libs/js-popup/package.json](https://github.com/cdnjs/cdnjs/pull/10975/files/5d9f739dcc37eef7894438890c9e3cb0f10fad4a#diff-1a8d227dee70c97cf53581b8c2c68f71) show, thanks.

File: ajax/libs/js-popup/package.json (hunk 6 above)
Comment (exelban) [inline ajax/libs/js-popup/package.json]: @sufuf3 in lib i have version for es6. In release for es5.

File: ajax/libs/js-popup/package.json (hunk 5 above)
Comment (PeterDaveHello) [inline ajax/libs/js-popup/package.json]: So clearly ask for removal.

**Comment by exelban** [comment]
@sufuf3 homepage removed. basePath must be release, because in lib i have version for es6.
And for what i must to change "*" --> "**/*"? 
In release folder i have only 3 main files.

**Comment by sufuf3** [comment]
`npmFileMap` is a list of files to take from the npm tarball
image
`files` is a pattern matcher allowing selection of multiple files, see [node-glob](https://github.com/isaacs/node-glob#glob-primer). use `"**/*"` is better.

**Comment by exelban** [comment]
@sufuf3 ohh. It takes files from npm. Im so sorry, its my fault.

**Comment by sufuf3** [comment]
@exelban Or would you like use git auto-update?
ref: https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

**Comment by exelban** [comment]
@sufuf3 another question. In npm i have version for es6 (with module.export).
I must to upload to npm and release version (like in PR)? If yes, i must to change npm version. Must i do the same for PR package.json?

**Comment by exelban** [comment]
@sufuf3 thanks. I will better change npm to autoupdate from github.

**Comment by sufuf3** [comment]
If you want to change auto-update from npm to GitHub.
you could delete as following:
```
  "npmName": "js-popup",
  "npmFileMap": [
    {
      "basePath": "release",
      "files": [
        "*"
      ]
    }
  ]
```
And add as the following:
```
  "autoupdate": {
    "source": "git",
    "target": "git://github.com/exelban/js-popup.git",
    "fileMap": [
      {
        "basePath": "release",
        "files": [
          "*/**"
        ]
      }
    ]
  }
```

**Comment by exelban** [comment]
@sufuf3 already

**Comment by exelban** [comment]
@sufuf3 must i change `
          "files": [ 
            "js-popup.js", 
            "js-popup.min.js", 
            "js-popup.css" 
          ] 
 ` to `
 "files": [
          "*/**"
        ]`?

**Comment by sufuf3** [comment]
@exelban please change it. `"*/**"` is more better.
And please use 2-spaces as indent, remove trailing spaces, you can use tool: tools/fixFormat.js

**Comment by exelban** [comment]
@sufuf3 how i can run this tool?

**Comment by sufuf3** [comment]
image

File: ajax/libs/js-popup/package.json (hunk 6 above)
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: sorry about that.
`npmFileMap` is a list of files to take from the npm tarball and host on cdnjs.
If use npm auto-update, it should modify as the following:
 `"basePath": "release",` --> `"basePath": "lib",`

File: ajax/libs/js-popup/package.json (hunk 7 above)
Comment (sufuf3) [inline ajax/libs/js-popup/package.json]: `files` is a pattern matcher allowing selection of multiple files, see [node-glob](https://github.com/isaacs/node-glob#glob-primer)
it  will be better to modify as the following:
`"*"` --> `"*/**"`

**Comment by exelban** [comment]
@sufuf3 i use gui. So i must to download repo to computer and run this tool?

**Comment by PeterDaveHello** [comment]
@sufuf3 can you please confirm that issuer is working locally or wanna work locally before asking that?

**Comment by PeterDaveHello** [comment]
@exelban sorry that @sufuf3 is one of our intern training students and sometimes she just didn't figure our the whole thing and respond too quick and may make something confusing. 

The tool runs on local, if you won't prefer to do that (as the whole repository is very huge), we can do it for you.

**Comment by exelban** [comment]
@PeterDaveHello its ok. Can you help me with CI tests?

**Comment by PeterDaveHello** [comment]
@exelban sure, I'll take it.

**Comment by PeterDaveHello** [comment]
@exelban should be done now, you just need to remove two trailing spaces :)

File: ajax/libs/js-popup/package.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    "url": "git+https://github.com/exelban/js-popup.git"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/exelban/js-popup.git",
+    "fileMap": [
+      {
+        "basePath": "release",
+        "files": [
+          "*/**"
```
Comment (PeterDaveHello) [inline ajax/libs/js-popup/package.json]: oops, I think here should be `"**/*"` not "*/**"

Comment (exelban) [inline ajax/libs/js-popup/package.json]: ohh, thank you very much.

**Comment by PeterDaveHello** [comment]
@cdnjs/intern3 : 
@extend1994 @sufuf3 @clairetsai818 @hare1039 please review and make sure everything good else except popularity and commit squashing

File: ajax/libs/js-popup/package.json (hunk 8 above)
Comment (extend1994) [inline ajax/libs/js-popup/package.json]: Remember to revise it :smiley:

File: ajax/libs/js-popup/package.json [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    "window-popup",
+    "popups",
+    "js-popup",
+    "component",
+    "JavaScript",
+    "box/popup window",
+    "box window",
+    "dialog window",
+    "dialog popup",
+    "dialog popup window"
```
Comment (extend1994) [inline ajax/libs/js-popup/package.json]: Here just suggestion for you.
There are many duplicated words such as `popup-window` and `popup-messages` which seemed not to let users be able to find the library more easily.
I tried some words in search engine like 
example
Maybe you can simplify it.

**Comment by sufuf3** [comment]
@exelban Would you like to finish this PR?
Thank you.

**Pull request closed without merging**